src/control/cmd/dmg/system.go

```diff
@@ -286,14 +286,20 @@ func displaySystemAction(log logging.Logger, results system.MemberResults,
 	}
 
 	if absentHosts.Count() > 0 {
-		out += fmt.Sprintf("\nUnknown %s: %s",
+		log.Infof("%s\nUnknown %s: %s", out,
 			english.Plural(absentHosts.Count(), "host", "hosts"),
 			absentHosts.String())
 	} else {
-		out += "\n"
+		log.Infof("%s\n", out)
 	}
 
-	log.Info(out)
+	if absentRanks.Count() > 0 || absentHosts.Count() > 0 {
+		return errors.New("non-existent ranks or hosts in cmd request")
+	}
+
+	if results.HasErrors() {
+		return errors.New("cmd results contain errors")
+	}
```

Comment:
While this is adequate for the short-term goal of indicating that a start/stop didn't succeed, it would be better to have the error include information about the failure. In the --json output case, everything logged to INFO is suppressed and the only way to learn what the problem was is to inspect the error field in the json output.

Reply:
well address in https://github.com/daos-stack/daos/pull/4795